src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```diff
@@ -134,6 +160,85 @@ public InputStream getJobLogFile(final int jobId, final JobLogInfo.Type logType)
         throw new UnsupportedOperationException("Job log info file retrieving operation haven't implemented yet");
     }
 
+    /**
+     * Creates the structure of an executable command based on the passed options.
+     *
+     * @param options User-defined options.
+     * @return The structure of an executable command.
+     */
+    private String[] makeSbatchCommand(final JobOptions options) {
+        final Context context = new Context();
+        context.setVariable(OPTIONS, options);
+        context.setVariable(LOG_DIR, logDir);
+        context.setVariable(ENV_VARIABLES, extractEnvVariablesFromOptions(options));
+        return commandCompiler.compileCommand(getProviderType(), SBATCH_COMMAND, context);
+    }
+
+    private void validateJobOptions(final JobOptions options) {
+        if (!StringUtils.hasText(options.getCommand())) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "Command should be specified!");
+        }
+        if (options.getPriority() < 0) {
+            throw new UnsupportedOperationException("Priority should be between 0 and 4_294_967_294");
+        }
+        if (options.getParallelEnvOptions() != null) {
+            throw new UnsupportedOperationException("Parallel environment variables are not supported yet!");
+        }
+        if (options.isCanBeBinary()) {
+            throw new UnsupportedOperationException("Scripts from command line are not supported yet!");
+        }
+    }
+
+    private String getResultOfExecutedCommand(final CmdExecutor cmdExecutor, final String[] command) {
+        final CommandResult result = cmdExecutor.execute(command);
+        if (result.getExitCode() != 0) {
+            CommandsUtils.throwExecutionDetails(result, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return parseJobId(result.getStdOut().get(0));
+    }
+
+    /**
+     * Gets the job ID from the string.
+     *
+     * @param jobString A string that can contain the job ID.
+     * @return JobID or an empty string.
+     */
+    private String parseJobId(final String jobString) {
+        final Pattern pattern = Pattern.compile(SUBMITTED_JOB_PATTERN);
```

Comment:
The pattern can be compiled during the class initialization in constructor or become a static member, because its expression is static and specified during startup time.

Reply:
done